float32 tempLeft
float32 tempRight
